### ISSUE TITLE
Unable to iterate over the entire results of policy_states.list_query_results_for_subscription.

### ISSUE THREAD

**Issue by ietashish** [Azure/azure-sdk-for-python#17280]
- **Package Name**: azure.mgmt.policyinsights
- **Package Version**: 1.0.0
- **Operating System**: MacOS
- **Python Version**: 3.7

**Describe the bug**
The issue happens when trying to list the policy state of entire subscription.

After instantiating `PolicyInsightsClient` and iterating over the results returned by  `client.policy_states.list_query_results_for_subscription(...)`, the following exception is thrown

``` 
 File "/Users/ashishg/****/venv/lib/python3.7/site-packages/azure/mgmt/policyinsights/operations/_policy_states_operations.py", line 352, in get_next
    raise HttpResponseError(response=response, model=error, error_format=ARMErrorFormat)
azure.core.exceptions.HttpResponseError: (HttpMethodNotSupported) The http method 'GET' is not supported for 'queryResults'.
```

Only the first 1000 policy state results are listed and when the client goes on to fetch the next 1000 results it errors out with the above exception.

The issue is happening because the Policy States List API (https://docs.microsoft.com/en-us/rest/api/policy/policystates/listqueryresultsforsubscription) expects the URL returned by `@odata.nextLink` attribute to be called over POST and not GET.

The problem seems to in line,
https://github.com/Azure/azure-sdk-for-python/blob/447e6f766ac809b95bac65e72752c44a59ea073f/sdk/policyinsights/azure-mgmt-policyinsights/azure/mgmt/policyinsights/operations/_policy_states_operations.py#L139

Things start to work, on changing the line from `request = self._client.get(url, query_parameters, header_parameters)`
to 
`request = self._client.post(url, query_parameters, header_parameters)`

**To Reproduce**
1. `pip install azure-mgmt-policyinsights` (latest version which is 1.0.0)
2. Run the following simple snippet to reproduce the issue. 
NOTE: Make sure that the subscription this is tried upon has more than 1000 policy state results.
```
from azure.mgmt.policyinsights._policy_insights_client import PolicyInsightsClient

#credential = <Create the credential object here>
#sub_id = <some subscription id>

client = PolicyInsightsClient(cred, base_url=mavlink_helpers.get_arm_base_url(tenant_region), subscription_id=sub_id)
policy_states = client.policy_states.list_query_results_for_subscription(policy_states_resource="latest",
                                                                             subscription_id=sub_id)
for policy_state in policy_states:
    policy_state = policy_state.as_dict()
    print(policy_state['policy_definition_id'], policy_state['resource_id'])

```

**Expected behavior**
One should be able to iterate over the entire list of policy states.

**Comment by yunhaoling** [comment]
thanks for the feedback! we'll investigate this asap.
looping in @msyyc who could provide further help on mgmt libraries.

**Comment by msyyc** [comment]
@00Kai0 please provide help on the issue

**Comment by 00Kai0** [comment]
This issue happened because next link using wrong method.
Normally, next_link will use `get` to get next page. But this API uses an unusual `post` request to get data.
https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/policyinsights/azure-mgmt-policyinsights/azure/mgmt/policyinsights/operations/_policy_states_operations.py#L139

**Comment by msyyc** [comment]
Fix swagger PR :https://github.com/Azure/azure-sdk-for-python/issues/17280

**Comment by SaurabhSharma-MSFT** [comment]
@ietashish I am following to know if your issue is resolved ? Please let us know.

**Comment by msyyc** [comment]
FYI: context for POST paging of Policy insights 


At that time, Python SDK received an issue [Unable to iterate over the entire results of policy_states.list_query_results_for_subscription. · I…](https://github.com/Azure/azure-sdk-for-python/issues/17280) that next link shall be accessed with POST instead of GET

<img width="1049" height="248" alt="Image" src="https://github.com/user-attachments/assets/5ee41c6a-a8a4-45af-8b61-7adbffce7cc2" />

After discussion, we decided to use tricky directive to fix it for Python SDK. Then we made 2 PRs [Fix to support nextLink with post calls for python sdk by najagasi · Pull Request #14035 · Azure/az…](https://github.com/Azure/azure-rest-api-specs/pull/14035) and [[policyinsights] Fix next_link by msyyc · Pull Request #15165 · Azure/azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs/pull/15165).



As for why I disable the the directive for other languages, accordign to the conversation of [[policyinsights] Fix next_link by msyyc · Pull Request #15165 · Azure/azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs/pull/15165), it is because the directive only work after adopting new autorest.core which other languages can't upgrade in short time.

<img width="800" height="513" alt="Image" src="https://github.com/user-attachments/assets/9545c24c-7085-4b8e-a4f6-4404291e1b6f" />
